File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskDeleteTask.java
package net.imatruck.dsmanager.tasks;

import android.app.Activity;
import android.os.AsyncTask;
import android.widget.TextView;

import net.imatruck.dsmanager.R;
import net.imatruck.dsmanager.models.DSTaskDeleteBase;
import net.imatruck.dsmanager.models.DSTaskDeleteData;
import net.imatruck.dsmanager.utils.SynologyDSTaskError;

import java.io.IOException;
import java.util.List;
import java.util.Locale;

import retrofit2.Call;

/**
 * Created by marc on 2017-03-24.
 */

public class DSTaskDeleteTask extends AsyncTask<Call<DSTaskDeleteBase>, Void, DSTaskDeleteBase> {

    private Activity context;

    public DSTaskDeleteTask(Activity context) {
        this.context = context;
    }

    @SafeVarargs
    @Override
    protected final DSTaskDeleteBase doInBackground(Call<DSTaskDeleteBase>... calls) {

        DSTaskDeleteBase dsTaskDeleteBase;
        try {
            dsTaskDeleteBase = calls[0].execute().body();
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        return dsTaskDeleteBase;
    }

    @Override
    protected void onPostExecute(DSTaskDeleteBase dsInfo) {
        if (dsInfo != null) {
            boolean success = dsInfo.isSuccess();
            TextView debugTextView = (TextView) context.findViewById(R.id.debug_api_text);

            if (success) {
                List<DSTaskDeleteData> infoData = dsInfo.getData();
                String text = "";

                for (DSTaskDeleteData dsTaskDeleteData : infoData) {
                    text += String.format(Locale.getDefault(),
                            "%s: %s\n",
                            dsTaskDeleteData.getId(),
                            context.getString(
                                    SynologyDSTaskError.getMessageId(dsTaskDeleteData.getError()))
                    );
                }
                debugTextView.setText(text);
            } else {
                String text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.getError().getCode()));
                debugTextView.setText(text);
            }
        }
    }
}

File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskDeleteTask.kt
package net.imatruck.dsmanager.tasks

import android.app.Activity
import android.os.AsyncTask
import android.widget.TextView
import net.imatruck.dsmanager.R
import net.imatruck.dsmanager.models.DSTaskDeleteBase
import net.imatruck.dsmanager.utils.SynologyDSTaskError
import retrofit2.Call
import java.io.IOException
import java.util.*


class DSTaskDeleteTask(private val context: Activity) : AsyncTask<Call<DSTaskDeleteBase>, Void, DSTaskDeleteBase>() {

    @SafeVarargs
    override fun doInBackground(vararg calls: Call<DSTaskDeleteBase>): DSTaskDeleteBase? {

        val dsTaskDeleteBase: DSTaskDeleteBase
        try {
            dsTaskDeleteBase = calls[0].execute().body()
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }

        return dsTaskDeleteBase
    }

    override fun onPostExecute(dsInfo: DSTaskDeleteBase?) {
        if (dsInfo != null) {
            val success = dsInfo.isSuccess
            val debugTextView = context.findViewById<TextView>(R.id.debug_api_text)

            if (success) {
                val infoData = dsInfo.data
                var text = ""

                for (dsTaskDeleteData in infoData) {
                    text += String.format(Locale.getDefault(),
                            "%s: %s\n",
                            dsTaskDeleteData.id,
                            context.getString(
                                    SynologyDSTaskError.getMessageId(dsTaskDeleteData.error))
                    )
                }
                debugTextView.text = text
            } else {
                val text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.error.code))
                debugTextView.text = text
            }
        }
    }
}
